lib/AST/Type.cpp

```diff
@@ -2426,6 +2429,11 @@ class IsBindableVisitor
       SmallVector<ProtocolConformanceRef, 4> paramSubstConformances;
       if (paramUpperBound) {
         for (auto proto : paramUpperBound->getConformsTo()) {
+          // FIXME: Skip invertible protocol requirements because they won't
+          // be in the substitution map.
+          if (proto->getInvertibleProtocolKind())
```

Comment:
@slavapestov 
This is the reduction of failing case I added as a test-case:

```
import Foundation

@objc protocol MyResult {
}

class Request<T : MyResult> {
}

struct Test {
  let closure: (Request<MyResult>) -> Void

  func test<R>(_ request: Request<R>) {
     self.closure(request as! Request<MyResult>)
  }
}
```

I don't know whether it's intentional or not but the generic signature of `decl` doesn't have `Copyable/Escapable` on `T` of `Request` (or at least it doesn't print them) which means we end up with a substitution map that does have these conformances but at the same time mapping generic parameter `T` into context results in an archetype that has Copyable/Escapable requirements. 

Even if we fix this in a different way and somehow add new conformances I think we'd still need to filter them in some situations based on the other place where are added the check.